examples/Huggingface_Transformers/Transformer_handler_generalized.py

```diff
@@ -75,12 +78,21 @@ def initialize(self, ctx):
                 self.model = AutoModelForCausalLM.from_pretrained(model_dir)
             else:
                 logger.warning("Missing the operation mode.")
-            self.model.to(self.device)
+            # HF GPT2 models options can be gpt2, gpt2-medium, gpt2-large, gpt2-xl
+            # this basically palce different model blocks on different devices,
+            # https://github.com/huggingface/transformers/blob/v4.17.0/src/transformers/models/gpt2/modeling_gpt2.py#L962
+            if self.setup_config["model_parallel"] and "gpt2" in self.setup_config["model_name"]:
+                self.model.parallelize() 
```

Comment:
How are we going to handle device_map?

Reply:
This has been handled [here](https://github.com/huggingface/transformers/blob/198c335d219a5eb4d3f124fdd1ce1a9cd9f78a9b/src/transformers/utils/model_parallel_utils.py#L50) if not passed explicitly.